CMakeLists.txt

```
@@ -112,6 +112,12 @@ endif()
 #
 # Third party
 #
+
+option(OFFLINE "Do not download anything from the internet" OFF)
```

Maybe default to `FETCHCONTENT_FULLY_DISCONNECTED`?
```suggestion
option(OFFLINE "Do not download anything from the internet" ${FETCHCONTENT_FULLY_DISCONNECTED})
```

or even omit this and use and document `FETCHCONTENT_FULLY_DISCONNECTED` instead as you are already using `FetchContent`